### ISSUE TITLE
custom SQL for ebpub.db.models.Location doesn't load during syncdb

### ISSUE THREAD

**Issue by slinkp** [openplans/openblock#74]
When running oblock sync_all, I notice this error (edited):

```
django-admin.py syncdb --settings=obdemo.settings --database=default --noinput
...
Creating table db_location
...
Installing custom SQL for db.Location model
Failed to install custom SQL for db.Location model: column "location" of relation "db_location" does not exist
...
Installing index for db.Location model
...
```

But the db_location column _does_ exist after this has all run, and the custom SQL (ebpub/ebpub/db/sql/location.sql) works fine if you run it manually after running oblock sync_all.

I can reproduce this in an existing build by recreating the databases:

```
oblock drop_dbs; oblock setup_dbs; oblock app=obdemo sync_all
```

The cause is that the column isn't actually created till the "installing index" step: see upstream bug at http://code.djangoproject.com/ticket/13826

Looking for a workaround that I can put in oblock.

**Comment by slinkp** [comment]
(In [79b68cb7a63bf76102b2bb2b2bd8c23e27eb0283]) comments, refs #74

**Comment by slinkp** [comment]
(In [bb93e5cf5a2722ba7726ddc88a207bcce8ca9e5e]) Fixes #95, fixes #74. Add a new signal and a handler for it that loads our custom location sql only after all the geom columns exist; use the intersecting_collection function in more places to avoid blowing up on GeometryCollections.

**Comment by slinkp** [comment]
Ticket imported from Trac:
 http://developer.openblockproject.org/ticket/74
Reported by: slinkp
